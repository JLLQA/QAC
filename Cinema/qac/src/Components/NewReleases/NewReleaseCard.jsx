import { Card, CardBody, CardText } from "reactstrap";

const NewReleaseCard = ({ newrel }) => {

    console.log(newrel);

    return (
        <Card >
            <CardBody>
                <img width="100" height="100" src={newrel.poster} alt="Card image cap" />
                <CardText>{newrel.id}</CardText>
                <CardText>{newrel.genre}</CardText>
                <CardText>{newrel.releasedate}</CardText>
            </CardBody>
        </Card>
    )

}

export default NewReleaseCard;