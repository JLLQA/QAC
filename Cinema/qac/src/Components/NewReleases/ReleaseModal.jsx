import { Modal, Container, Row, Col, Button } from "react-bootstrap";


const ReleaseModal = (props) => {

    const eachRel = props.data;
    console.log(eachRel);

    return (
        <Modal {...props} aria-labelledby="contained-modal-title-vcenter" size="lg" centered >
            <Modal.Header closeButton>
                <Modal.Title id="contained-modal-title-vcenter"  >
                    {eachRel.title}
                </Modal.Title>
            </Modal.Header>
            <Modal.Body className="show-grid" >
                <Container >
                    <Row>
                        <Col xs={6} md={4}>
                            <Row>
                            Main Actor: {eachRel.actors}
                            </Row>
                            <Row>
                            Main Director: {eachRel.director}
                            </Row>
                            <Row>
                            Expected Release: {eachRel.releasedate}
                            </Row>
                        </Col>
                        <Col >
                            <img src={eachRel.poster} alt={eachRel.title} align="right" />
                        </Col>
                    </Row>

                    <Row>
                           Plot: {eachRel.story}
                    </Row>
                </Container>
            </Modal.Body>
            <Modal.Footer>
                <Button  onClick={props.onHide} align="left">Close</Button>
            </Modal.Footer>
        </Modal>
    );
}

export default ReleaseModal;