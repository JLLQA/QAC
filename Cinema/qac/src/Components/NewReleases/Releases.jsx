import NewReleaseCard from "./NewReleaseCard";
import Searchbar from "../Searchbar/Search";

const Releases = ({ props }) => {

    const { data, query, q } = props;

    // //searchBar methods
    const filterQuery = () => (data) => data.title.toLowerCase().includes(query.toLowerCase());
    let filteredList = data.filter(filterQuery());
    

    //Return body
    return (
        <>
        <Searchbar query={q}/>
         {
            filteredList.map((newrel)=>(
                
            <NewReleaseCard key={newrel.id} newReleases={newrel}/>
            ))
        } 
        </>

    )

}

export default Releases;