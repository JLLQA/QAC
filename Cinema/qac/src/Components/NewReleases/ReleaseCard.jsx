import { useState } from "react";
import ReleaseModal from "./ReleaseModal";

const ReleaseCard = ({ release }) => {
    console.log(release);

    const [modalShow, setModalShow] = useState(false);
    return (
        <div className="nav-links" style={{ marginBottom: "1%", marginTop: "1%", marginRight: "1%", marginLeft: "1%", border: "2px solid #ffccff", borderRadius: "5px", width:"90%" }}>
            <div onClick={() => setModalShow(true)} style={{ cursor:"pointer"}}>
                <h3>{release.title}</h3>
                <img src={release.poster} alt={release.title} />
                <p>Expected Release Date: {release.releasedate}</p>
            </div>
            <ReleaseModal show={modalShow} onHide={() => setModalShow(false)} data={release} />
        </div>
    )
}
export default ReleaseCard;