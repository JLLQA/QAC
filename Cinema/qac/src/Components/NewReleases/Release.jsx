// import { Container } from "reactstrap";
import ReleaseCard from "./ReleaseCard";

const Release = ({ data , query}) => {
    console.log(data);
    
    const filterQuery = () => (data) => data.title.toLowerCase().includes(query.toLowerCase());
    let filteredList = data.filter(filterQuery());

    return (
          <div>      
         {
            filteredList.map((rel)=>(
            <ReleaseCard key={rel._id} release={rel}/>
            ))
        } 
        </div>
    )

}

export default Release;