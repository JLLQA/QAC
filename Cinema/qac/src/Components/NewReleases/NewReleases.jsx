import axios from "axios";
import { useEffect, useState } from "react";
import Navbar from "../Multipage/Navbar/Navbar";
import Release from "./Release";


const NewReleases = () => {

    const [data, setData] = useState([]);
    const [isLoaded, setIsLoaded] = useState(false);

    const [q, setQ] = useState("");

    const qHandler = (e) => {
        setQ(e.target.value);
    };

    useEffect(() => {
        axios({
            method: "Get",
            url: "http://127.0.0.1:5000/newreleases",
            headers: { "Access-Control-Allow-Origin": "*" }
        })
            .then((resp) => {
                // console.log(resp.data);
                setData(resp.data);
                setIsLoaded(true);
            })
            .catch((err) => {
                console.log(err.message);
                setIsLoaded(true);
            });
    }, []);

    if (isLoaded) {
        return (
            <div>
                <Navbar />
                {/* <h1>NEW RELEASES</h1> */}
                <div>
                        <Release data={data} query={q}/>  
                </div>
                
            </div>
        );
    }
    else {
        return (
            <div>
                <Navbar />
                <h1>Loading...</h1>
            </div>
        );
    }

}

export default NewReleases;