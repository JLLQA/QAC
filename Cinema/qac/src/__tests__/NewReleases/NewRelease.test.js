import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import NewReleases from '../../Components/NewReleases/NewReleases';
import renderer from 'react-test-renderer';




describe('Test for NewReleases page', () => {

    test('Check the NewRelease loading text is present', () => {
        render(<NewReleases />);
        const linkElement = screen.getByText("Loading...");
        expect(linkElement).toBeInTheDocument();
    });

    test("It matches the snapshot", () => {
        const tree = renderer.create(<NewReleases />).toJSON();
        expect(tree).toMatchSnapshot();
    });



})