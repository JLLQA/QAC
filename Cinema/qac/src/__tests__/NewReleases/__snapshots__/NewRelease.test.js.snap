// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for NewReleases page It matches the snapshot 1`] = `
<div
  className="container-fluid"
  id="dropped-box"
>
  <h1>
    Loading...
  </h1>
</div>
`;
